pulp_ansible/tests/functional/cli/test_collection_install.py

```diff
@@ -77,3 +83,96 @@ def test_install_collection(self):
             subprocess.run(cmd.split())
 
             self.assertTrue(path.exists(directory), "Could not find directory {}".format(directory))
+
+    @skip_if(bool, "AH_token", False)
+    def test_install_collection_with_token_from_automation_hub(self):
+        """Test whether ansible-galaxy can install a Collection hosted by Pulp."""
+        repo = self.repo_api.create(gen_repo())
+        self.addCleanup(self.repo_api.delete, repo.pulp_href)
+
+        body = gen_ansible_remote(
+            url=TOKEN_AUTH_COLLECTION_TESTING_URL,
+            auth_url=AH_AUTH_URL,
+            token=os.environ["AUTOMATION_HUB_TOKEN_AUTH"],
+            tls_validation=False,
+        )
+        remote = self.remote_collection_api.create(body)
+        self.addCleanup(self.remote_collection_api.delete, remote.pulp_href)
+
+        # Sync the repository.
+        self.assertEqual(repo.latest_version_href, f"{repo.pulp_href}versions/0/")
+        repository_sync_data = RepositorySyncURL(remote=remote.pulp_href)
+        sync_response = self.repo_api.sync(repo.pulp_href, repository_sync_data)
+        monitor_task(sync_response.task)
+        repo = self.repo_api.read(repo.pulp_href)
+
+        # Create a distribution.
+        body = gen_distribution()
+        body["repository"] = repo.pulp_href
+        distribution_create = self.distributions_api.create(body)
+        distribution_url = monitor_task(distribution_create.task)
+        distribution = self.distributions_api.read(distribution_url[0])
+
+        self.addCleanup(self.distributions_api.delete, distribution.pulp_href)
+
+        with tempfile.TemporaryDirectory() as temp_dir:
+            cmd = "ansible-galaxy collection install {} -c -s {} -p {}".format(
+                TOKEN_DEMO_COLLECTION, distribution.client_url, temp_dir
+            )
+
+            directory = "{}/ansible_collections/{}".format(
+                temp_dir, TOKEN_DEMO_COLLECTION.replace(".", "/")
+            )
+
+            self.assertTrue(
+                not path.exists(directory), "Directory {} already exists".format(directory)
+            )
+
+            subprocess.run(cmd.split())
+
+            self.assertTrue(path.exists(directory), "Could not find directory {}".format(directory))
+
+    @skip_if(bool, "GH_token", False)
+    def test_install_collection_with_token_from_galaxy(self):
```

Comment:
It is more like a placeholder, as I don't know which collection requires token auth to be synced, I put `pulp.pulp_installer` here.
@bmbouter @ironfroggy @chouseknecht